src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -561,7 +561,8 @@
   [{:keys [server] :as webserver-context} :- ServerContext]
   (when-let [mbean-container (:mbean-container @(:state webserver-context))]
     (log/debug "Cleaning up JMX MBean container")
-    (.destroy mbean-container))
+    (.destroy mbean-container)
+    (swap! (:state webserver-context) assoc :mbean-container nil))
```

Comment:
If `shutdown` were being called from two different threads at the same time, it seems possible that both threads could have gotten the `:mbean-container` and subsequently try to call `.destroy` on it.  There's probably another way to do this that would involve only calling `.destroy` here if we've successfully moved the current state of the atom from having `:mbean-container` set with the original object to having `:mbean-container` not have that object anymore.  Although if we're eventually going to preclude the possibility of this function being callable from two different threads at the same time, what you already have here is probably sufficient.